src/Glpi/Exception/Access/InvalidCsrfException.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+/**
+ * ---------------------------------------------------------------------
+ *
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ *
+ * http://glpi-project.org
+ *
+ * @copyright 2015-2024 Teclib' and contributors.
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+namespace Glpi\Exception\Access;
+
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpFoundation\StreamedResponse;
+
+class InvalidCsrfException extends AccessException
+{
+    public function asResponse(): Response
+    {
+        $message = __("The action you have requested is not allowed.");
+
+        $request = $this->getRequest();
+
+        // Output JSON if requested by client
+        if (str_contains($request->getAcceptableContentTypes()['HTTP_ACCEPT'] ?? '', 'application/json')) {
+            return new JsonResponse(['message' => $this->message], 403);
+        }
+
+        return new StreamedResponse(function () use ($message) {
+            \Html::displayError($message, true);
+        }, 403);
+    }
```

Comment:
From the end user point of view, the only difference with an uncatched exception will be the displayed message.

Maybe switch/case blocks could be added in the `Glpi\Controller\ErrorController` methods to adapt the message / HTTP code returned in the response. To handle this case, the `asResponse()` signature could be changed to be able to return `null`.

Also, for this specific case, maybe a `Symfony\Component\HttpKernel\Exception\AccessDeniedHttpException` can be thrown directly. Indeed, the message and the HTTP code returned here is pretty generic.

Reply:
For this part, throwing instances of `HttpExceptionInterface` is enough: we have a message and a status code. Maybe I'll change the Session class to use it instead.

I will remove the listener and do the work directly in the `ErrorController`: it's more straightforward.